db/deprecated/utils.py

```
@@ -21,14 +21,17 @@ def execute_statement(engine, statement, connection_to_use=None):
             with engine.begin() as conn:
                 return conn.execute(statement)
     except ProgrammingError as e:
-        if isinstance(e.orig, p_errors.UndefinedFunction):
+        if isinstance(e.orig, PsycopgUndefinedFunction):
             message = e.orig.args[0].split('\n')[0]
             raise UndefinedFunction(message)
```

Ruff (0.15.20)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)